sys/src/runtime.rs

```diff
@@ -14,3 +18,12 @@ extern "C" {
     pub fn caml_gc_full_major(v: Value);
     pub fn caml_gc_compaction(v: Value);
 }
+
+#[cfg(ocaml5)]
+extern "C" {
+    pub fn caml_init_domain_self(i: core::ffi::c_int);
+    pub fn caml_interrupt_self();
+    pub fn caml_reset_domain_lock();
+    pub fn caml_acquire_domain_lock();
```

Comment:
I've stumbled across this discuss topic while searching for `caml_acquire_domain_lock()`, according to the discussion - one should not use functions from `domain.h` directly 🤔